test/extended/util/client.go

```diff
@@ -323,14 +324,44 @@ func (c *CLI) setupProject() string {
 	// Wait for SAs and default dockercfg Secret to be injected
 	// TODO: it would be nice to have a shared list but it is defined in at least 3 place,
 	// TODO: some of them not even using the constants
+	// deployer and builder is not must for some clusters, such as no-capabilities
 	DefaultServiceAccounts := []string{
 		"default",
-		"deployer",
-		"builder",
 	}
+	shouldCheckSecret := false
+	clusterVersion, err := c.AdminConfigClient().ConfigV1().ClusterVersions().Get(context.Background(), "version", metav1.GetOptions{})
+	o.Expect(err).NotTo(o.HaveOccurred())
+	checkCapability := func(capabilities []configv1.ClusterVersionCapability, checked configv1.ClusterVersionCapability) bool {
+		m := ConvertStrSlice2Map(capabilities)
+		_, ok := m[checked]
+		if ok {
+			return true
+		} else {
+			return false
+		}
+	}
+	if clusterVersion.Status.Capabilities.KnownCapabilities == nil ||
+		!checkCapability(clusterVersion.Status.Capabilities.KnownCapabilities, configv1.ClusterVersionCapabilityBuild) ||
+		(clusterVersion.Status.Capabilities.EnabledCapabilities != nil &&
+			checkCapability(clusterVersion.Status.Capabilities.EnabledCapabilities, configv1.ClusterVersionCapabilityBuild)) {
```

Comment:
this whole condition is repeated below for `DeploymentConfig` and `ImageRegistry`.  Maybe expand `checkCapability` to handle the whole thing?  And shift it out to a standalone function, so the function definition doesn't distract `setupProject` readers?

```go
if capabilityEnabled(clusterVersion.Status.Capabilities, configv1.ClusterVersionCapabilityBuild) {
  DefaultServiceAccounts = append(DefaultServiceAccounts, "builder")
}
if capabilityEnabled(clusterVersion.Status.Capabilities, configv1.ClusterVersionCapabilityDeploymentConfig) {
  DefaultServiceAccounts = append(DefaultServiceAccounts, "deployer")
}
// ...
```

Reply:
Updated, please have a review when you get a chance, thanks!